### PR TITLE
refactor(client): add return types

### PR DESCRIPTION
**Pull request by tarunama**
## summary
- Explicitly define return types
- Add type of  [Observer](https://github.com/vercel/next.js/pull/20728/files#diff-5de64b97b2f26e4e41d197a8295e8750825c75b8ca557a4b947a4c3569345974R7)

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| buildDuration | 10.6s | 11.2s | ⚠️ +552ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ⚠️ +574 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.221 | 2.259 | ⚠️ +0.04 |
| / avg req/sec | 1125.67 | 1106.62 | ⚠️ -19.05 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.585 | 1.534 | -0.05 |
| /error-in-render avg req/sec | 1576.86 | 1629.59 | +52.73 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| 677f882d2ed8..42af.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-a9a6f0d..a96d.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| index.html gzip | 615 B | 615 B | ✓ |
| link.html gzip | 621 B | 620 B | -1 B |
| withRouter.html gzip | 607 B | 607 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | -1 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-21a66e0..022b3306c.js</strong></summary>

```diff
@@ -1730,6 +1730,7 @@ _N_E = (window["webpackJsonp_N_E"] = window["webpackJsonp_N_E"] || []).push([
             /**
              * @param {string} href the route href (file-system path)
              * @param {string} asPath the URL as shown in browser (virtual path); used for dynamic routes
+             * @returns {string}
              */
           },
           {
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-21a66e0d155022b3306c.js"
+      href="/_next/static/chunks/main-675149f0436ec1306827.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-21a66e0d155022b3306c.js"
+      src="/_next/static/chunks/main-675149f0436ec1306827.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-21a66e0d155022b3306c.js"
+      href="/_next/static/chunks/main-675149f0436ec1306827.js"
       as="script"
     />
     <link
@@ -58,7 +58,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-21a66e0d155022b3306c.js"
+      src="/_next/static/chunks/main-675149f0436ec1306827.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-21a66e0d155022b3306c.js"
+      href="/_next/static/chunks/main-675149f0436ec1306827.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-21a66e0d155022b3306c.js"
+      src="/_next/static/chunks/main-675149f0436ec1306827.js"
       async=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| buildDuration | 13s | 13.3s | ⚠️ +267ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ⚠️ +574 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| 677f882d2ed8..42af.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-a9a6f0d..a96d.js gzip | 6.59 kB | N/A | N/A |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| main-93f263e..9853.js gzip | N/A | 6.59 kB | N/A |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.17 MB | 5.17 MB | ✓ |

</details>

</details>
Commit: 5263867ffdaf7ae6ccfd8e9c44178bda837862f8

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| buildDuration | 10.2s | 9.9s | -298ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ⚠️ +574 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.998 | 1.994 | 0 |
| / avg req/sec | 1250.99 | 1253.88 | +2.89 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.231 | 1.235 | 0 |
| /error-in-render avg req/sec | 2030.58 | 2023.63 | ⚠️ -6.95 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| 677f882d2ed8..396f.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-6b712d3..3360.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| index.html gzip | 615 B | 614 B | -1 B |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 607 B | -1 B |
| Overall change | 1.84 kB | 1.84 kB | -2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-c371cf8..efe26c758.js</strong></summary>

```diff
@@ -1739,6 +1739,7 @@ _N_E = (window["webpackJsonp_N_E"] = window["webpackJsonp_N_E"] || []).push([
             /**
              * @param {string} href the route href (file-system path)
              * @param {string} asPath the URL as shown in browser (virtual path); used for dynamic routes
+             * @returns {string}
              */
           },
           {
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      href="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      src="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      href="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       as="script"
     />
     <link
@@ -58,7 +58,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      src="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      href="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      src="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       async=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| buildDuration | 12s | 11.8s | -192ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ⚠️ +574 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| 677f882d2ed8..396f.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-6b712d3..3360.js gzip | 6.63 kB | N/A | N/A |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| main-d74fce9..a1a1.js gzip | N/A | 6.63 kB | N/A |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.17 MB | 5.17 MB | ✓ |

</details>

</details>
Commit: f9b774dcefef945dd074ee48a32d1864dfbfde59

File: packages/next/client/route-loader.ts [hunk 1]
```diff
@@ -166,7 +166,9 @@ export function getClientBuildManifest(): Promise<ClientBuildManifest> {
     return Promise.resolve(self.__BUILD_MANIFEST)
   }
 
-  const onBuildManifest = new Promise<ClientBuildManifest>((resolve) => {
+  const onBuildManifest: Promise<Record<string, string[]>> = new Promise<
```
Comment (Timer) [inline packages/next/client/route-loader.ts:169]:
```suggestion
  const onBuildManifest: Promise<ClientBuildManifest> = new Promise<
```

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]
Thanks!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| buildDuration | 10s | 9.8s | -160ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ⚠️ +569 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.048 | 1.952 | -0.1 |
| / avg req/sec | 1220.46 | 1280.76 | +60.3 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.234 | 1.216 | -0.02 |
| /error-in-render avg req/sec | 2025.76 | 2056.03 | +30.27 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| 677f882d2ed8..396f.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-6b712d3..3360.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| index.html gzip | 615 B | 614 B | -1 B |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 608 B | 607 B | -1 B |
| Overall change | 1.84 kB | 1.84 kB | -2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>main-c371cf8..efe26c758.js</strong></summary>

```diff
@@ -1739,6 +1739,7 @@ _N_E = (window["webpackJsonp_N_E"] = window["webpackJsonp_N_E"] || []).push([
             /**
              * @param {string} href the route href (file-system path)
              * @param {string} asPath the URL as shown in browser (virtual path); used for dynamic routes
+             * @returns {string}
              */
           },
           {
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      href="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      src="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      href="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       as="script"
     />
     <link
@@ -58,7 +58,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      src="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      href="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-ae47a1f95297af5f55f8.js"
     ></script>
     <script
-      src="/_next/static/chunks/main-c371cf8cfc0efe26c758.js"
+      src="/_next/static/chunks/main-500e5a24a4b6ddc750ae.js"
       async=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| buildDuration | 12s | 11.9s | -127ms |
| nodeModulesSize | 80.6 MB | 80.6 MB | ⚠️ +569 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| 677f882d2ed8..396f.js gzip | 13 kB | 13 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-6b712d3..3360.js gzip | 6.63 kB | N/A | N/A |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| main-d74fce9..a1a1.js gzip | N/A | 6.63 kB | N/A |
| Overall change | 59.3 kB | 59.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-705099c..c35d.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | tarunama/next.js chore/add-return-types | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.17 MB | 5.17 MB | ✓ |

</details>

</details>
Commit: e8bca328721286e2f6c046302813d8000f735ead

**Pull request merged**